### PR TITLE
tsfn: add error checking on GetContext

### PR DESCRIPTION
**Pull request by KevinEady**
Fixes: https://github.com/nodejs/node-addon-api/issues/581

### CODE REVIEW

**Comment by KevinEady** [comment]
https://github.com/nodejs/node-addon-api/blob/e142a1ee57e5ebf8030fbd174ae2d481b727494b/napi-inl.h#L4113-L4114

Will need to review if FATAL error is correct. If not, how to properly handle this error

File: test/threadsafe_function/threadsafe_function_ctx.cc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  Reference<Napi::Value> *_ctx = new Reference<Napi::Value>;
+  *_ctx = Persistent(info[0]);
+
+  _tsfn = ThreadSafeFunction::New(
+      info.Env(), Function::New(env, [](const CallbackInfo & /*info*/) {}),
+      Object::New(env), "Test", 1, 1, _ctx,
+      [this](Napi::Env env, Reference<Napi::Value> *ctx) {
+        _deferred->Resolve(env.Undefined());
+        _deferred.reset();
```
Comment (legendecas) [inline test/threadsafe_function/threadsafe_function_ctx.cc]: I don't think reset is necessary here as the unique ptr would call the deleter on destruct.

Comment (KevinEady) [reply test/threadsafe_function/threadsafe_function_ctx.cc]: :+1: will update in later push, when master is updated with #583 since this PR conflicts

**Comment by KevinEady** [comment]
Hey @mhdawson  / @gabrielschulhof ,

On the same basis of #566 (see https://github.com/nodejs/node-addon-api/pull/566#issuecomment-544637060) , would this also be considered SemVer major? If so, should we put this in to the 2.0.0 release (the scope seems to be ever increasing...)

In this case, "should not affect any code": If you were using `GetContext()` after an error and you still used the resulting `Context*` pointer, you'd probably crash with an error anyway?

This PR also directly conflicts with #580 because of the modifications of the test files, and as 580 is higher prio, I can't update this until that PR is merged.

**Comment by mhdawson** [comment]
@KevinEady do you know when you will be able to push the change to address the outstanding comment?

File: test/threadsafe_function/threadsafe_function_ctx.cc (hunk 1 above)
Comment (KevinEady) [reply test/threadsafe_function/threadsafe_function_ctx.cc]: Updated

**Comment by KevinEady** [comment]
Hi @mhdawson , just pushed!

**Comment by mhdawson** [comment]
Landed as c881168d4951e833fb8275b233fa181ec0b60b43

**Pull request merged**